packages/amplify_authenticator/lib/src/mixins/authenticator_phone_field.dart

```diff
@@ -1,153 +1,158 @@
+  Future<void> showCountryDialog() async {
+    // Reset search
+    _searchVal = '';
 
-    Future<void> showCountryDialog() async {
-      await showDialog<Country>(
-          context: context,
-          builder: (context) {
-            return SimpleDialog(
-              title: Text(countryResolver.resolve(
-                  context, CountryResolverKey.selectDialCode)),
-              children: <Widget>[
-                Container(
-                    padding: const EdgeInsets.fromLTRB(20, 0, 20, 0),
+    final selectedCountry = await showDialog<Country>(
+      context: context,
+      builder: (context) {
+        return StatefulBuilder(
+          builder: (context, setState) {
+            return Dialog(
+              child: Column(
+                children: <Widget>[
+                  Padding(
+                    padding: const EdgeInsets.all(24.0),
+                    child: Text(
+                      _countriesResolver.resolve(
+                        context,
+                        CountryResolverKey.selectDialCode,
+                      ),
+                      style: DialogTheme.of(context).titleTextStyle ??
+                          Theme.of(context).textTheme.headline6!,
+                    ),
+                  ),
+                  Padding(
+                    padding: const EdgeInsets.symmetric(horizontal: 20),
                     child: TextField(
-                      decoration:
-                          const InputDecoration(suffixIcon: Icon(Icons.search)),
-                      onChanged: (String newValue) {
-                        /// Make sure we search on full list when search string gets shorter
-                        if (_searchVal.length > newValue.length) {
-                          setState(() {
-                            _countries.value = countryCodes;
-                          });
-                        }
+                      decoration: const InputDecoration(
+                        suffixIcon: Icon(Icons.search),
+                        isDense: true,
+                      ),
+                      onChanged: (String searchVal) {
                         setState(() {
-                          /// Filter the list of countries based on localized display values
-                          _searchVal = newValue;
-
-                          /// If search string is empty, display full list; otherwise, filter
-                          newValue.isEmpty
-                              ? _countries.value = countryCodes
-                              : _countries.value = _countries.value
-                                  .where((c) => countryResolver
-                                      .resolve(context, c.key)
-                                      .toLowerCase()
-                                      .contains(newValue.toLowerCase()))
-                                  .toList();
+                          _searchVal = searchVal;
```

Comment:
When I did the initial implementation I tried to use a TextController for this, but  Flutter would throw an exception about an invalid key (similar to the one we we fixed earlier). Did you see this as well?

Reply:
I can't remember the earlier issue, but I haven't had any problems with this.